core/chains/evm/gas/universal_estimator.go

```diff
@@ -0,0 +1,413 @@
+		bumpedMaxPriorityFeePerGas = assets.WeiMin(bumpedMaxPriorityFeePerGas, maxPrice)
+	}
+
+	bumpedMaxFeePerGas, err = LimitBumpedFee(originalFee.FeeCap, currentDynamicPrice.FeeCap, bumpedMaxFeePerGas, maxPrice)
+	if err != nil {
+		return bumped, fmt.Errorf("maxFeePerGas error: %s", err.Error())
+	}
+
+	bumpedFee := DynamicFee{FeeCap: bumpedMaxFeePerGas, TipCap: bumpedMaxPriorityFeePerGas}
+	u.logger.Debugw("bumped dynamic fee", "originalFee", originalFee, "bumpedFee", bumpedFee)
+
+	return bumpedFee, nil
+}
+
+// LimitBumpedFee selects the maximum value between the bumped attempt and the current fee, if there is one. If the result is higher than the max price it gets capped.
+// Geth's implementation has a hard 10% minimum limit for the bumped values, otherwise it rejects the transaction with an error.
+// See: https://github.com/ethereum/go-ethereum/blob/bff330335b94af3643ac2fb809793f77de3069d4/core/tx_list.go#L298
+//
+// Note: for chains that support EIP-1559 but we still choose to send Legacy transactions to them, the limit is still enforcable due to the fact that Legacy transactions
+// are treated the same way as Dynamic transactions under the hood. For chains that don't support EIP-1559 at all, the limit isn't enforcable but a 10% minimum bump percentage
+// makes sense anyway.
+func LimitBumpedFee(originalFee *assets.Wei, currentFee *assets.Wei, bumpedFee *assets.Wei, maxPrice *assets.Wei) (*assets.Wei, error) {
+	if currentFee != nil {
+		bumpedFee = assets.WeiMax(currentFee, bumpedFee)
+	}
+	bumpedFee = assets.WeiMin(bumpedFee, maxPrice)
+
+	// The first check is added for the following edge case:
+	// If originalFee is below 10 wei, then adding the minimum bump percentage won't have any effect on the final value because of rounding down.
+	// Similarly for bumpedFee, it can have the exact same value as the originalFee, even if we bumped, given an originalFee of less than 10 wei
+	// and a small enough BumpPercent.
+	if bumpedFee.Cmp(originalFee) == 0 ||
+		bumpedFee.Cmp(originalFee.AddPercentage(MinimumBumpPercentage)) < 0 {
+		return nil, fmt.Errorf("%w: %s is bumped less than minimum allowed percentage(%s) from originalFee: %s - maxPrice: %s",
+			commonfee.ErrBump, bumpedFee, strconv.Itoa(MinimumBumpPercentage), originalFee, maxPrice)
+	}
+	return bumpedFee, nil
+}
+
+func (u *UniversalEstimator) getPriorityFeeThreshold() (*assets.Wei, error) {
+	u.priorityFeeThresholdMu.RLock()
+	defer u.priorityFeeThresholdMu.RUnlock()
+	if u.priorityFeeThreshold == nil {
+		return u.priorityFeeThreshold, fmt.Errorf("priorityFeeThreshold not set")
+	}
+	return u.priorityFeeThreshold, nil
+}
+
+func (u *UniversalEstimator) Name() string                                      { return u.logger.Name() }
+func (u *UniversalEstimator) L1Oracle() rollups.L1Oracle                        { return u.l1Oracle }
+func (u *UniversalEstimator) HealthReport() map[string]error                    { return map[string]error{u.Name(): nil} }
```

Comment:
Should not we mark service as unhealthy if there is no iteration of `run` that completed successfully?
And if refresh of the cache fails.
Otherwise why do we need `HealthReport()`?

Reply:
`HealthReport` was added just because it is required by the Service interface, and I haven't seen being utilized by our components so far, that's why I didn't add a check.